test/integration/original_ip_detection_integration_test.cc

```diff
@@ -0,0 +1,44 @@
+#include "envoy/extensions/original_ip_detection/custom_header/v3/custom_header.pb.h"
+
+#include "test/integration/http_integration.h"
+#include "test/test_common/registry.h"
+#include "test/test_common/utility.h"
+
+#include "gtest/gtest.h"
+
+using testing::HasSubstr;
+
+namespace Envoy {
+namespace Formatter {
+
+class OriginalIPDetectionIntegrationTest : public testing::Test, public HttpIntegrationTest {
```

Comment:
optional, I'd be inclined to fold this into integration_test.cc just because it's one less thing to link.

Reply:
passing on this one because a) that file is already quite long and b) it'd need to depend on http_integration.h so I think maybe it doesn't justify the saved time on linking (which granted, is quite a bit).